### PR TITLE
Remove Unnecessary Code From `casper_engine_test_support`.

### LINKED ISSUE #1978: Make the internal types in EE integration tests public

### PR DESCRIPTION
**Pull request by nilz3ro**
# About These Changes
This PR removes unnecessary code from the `casper_engine_test_support` crate.
* Deleted  `execution_engine_testing/test_support/src/internal/account.rs` and replaced any reference to `casper_engine_test_support::internal::Account` with `casper_types::account::Account`.
* Deleted `execution_engine_testing/test_support/src/internal/code.rs`.
* Deleted `execution_engine_testing/test_support/src/internal/error.rs`.
* Deleted `execution_engine_testing/test_support/src/internal/value.rs` and replaced call sites of `Value::into_t()` to use `CLValue::into_t()`.
* Deleted `execution_engine_testing/test_support/src/internal/session.rs`.
* Deleted `execution_engine_testing/test_support/src/internal/test_context.rs` and replaced every use of `TestContext` with an `InMemoryWasmTestBuilder`, using `DeployItemBuilder` and `ExecuteRequestBuilder`.
* Moved remaining files in `execution_engine_testing/test_support/src/internal` to `execution_engine_testing/test_support/src` and reexported them directly from `execution_engine_testing/test_support/src/lib.rs`.
* Removed call sites of `WasmTestBuilder::finish()`. 
* Updated all references to any type or module in `casper_engine_test_support::internal` to its new location in `casper_engine_test_support`.
* closes #1978

### CODE REVIEW

File: execution_engine_testing/test_support/Cargo.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ num-rational = "0.4.0"
 num-traits = "0.2.10"
 once_cell = "1.5.2"
 rand = "0.8.3"
+thiserror = "1.0.29"
```
Comment (nilz3ro) [inline execution_engine_testing/test_support/Cargo.toml]: remove this since we won't be using it.

File: Makefile [hunk 2]
```diff
@@ -119,7 +119,7 @@ lint: lint-contracts-rs
 
 .PHONY: audit
 audit:
-	$(CARGO) audit
+	$(CARGO) audit --ignore RUSTSEC-2021-0073 --ignore RUSTSEC-2021-0076
```
Comment (nilz3ro) [inline Makefile]: remove this if it's invalid / outdated.

File: execution_engine_testing/test_support/src/execute_request_builder.rs [hunk 3]
```diff
@@ -10,55 +10,63 @@ use casper_types::{
...
 };
 
-use crate::internal::{
-    DeployItemBuilder, DEFAULT_BLOCK_TIME, DEFAULT_PAYMENT, DEFAULT_PROPOSER_PUBLIC_KEY,
-};
+use crate::{DeployItemBuilder, DEFAULT_BLOCK_TIME, DEFAULT_PAYMENT, DEFAULT_PROPOSER_PUBLIC_KEY};
 
 const ARG_AMOUNT: &str = "amount";
 
+/// TODO: doc comment.
```
Comment (nilz3ro) [inline execution_engine_testing/test_support/src/execute_request_builder.rs]: Replace all occurrences of `TODO: doc comment.` with real doc comments.

**Comment by Fraser999** [comment]
* remove `thiserror` from Cargo.toml
* remove ` --ignore RUSTSEC-2021-0073 --ignore RUSTSEC-2021-0076` from Makefile
* remove "internal" folder
* remove session.rs
* remove test_context.rs
* remove `pub type URefAddr` and `pub type Hash` from lib.rs
* remove `WasmTestResult`
* rename 3 `..TestContext` back to `...TestBuilder`
* remove the lib's main doc comment example
* remove any `#[allow(deprecated)]` which were added but are not required any more

File: execution_engine_testing/test_support/src/deploy_item_builder.rs [hunk 4]
```diff
@@ -43,15 +47,18 @@ impl DeployItemBuilder {
         self
     }
 
+    /// Sets the payment bytes of the deploy to an empty Vec.
     pub fn with_empty_payment_bytes(self, args: RuntimeArgs) -> Self {
         self.with_payment_bytes(vec![], args)
     }
 
+    /// Sets the payment bytes of a deploy by reading a file and .
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/deploy_item_builder.rs]: Needs finished

File: execution_engine_testing/test_support/src/lib.rs [hunk 5]
```diff
@@ -65,35 +7,156 @@
...
-
-/// The address of a [`URef`](casper_types::URef) (unforgeable reference) on the network.
-pub type URefAddr = [u8; 32];
-
-/// The hash of a smart contract stored on the network, which can be used to reference the contract.
-pub type Hash = [u8; 32];
-
-/// Default test account address.
-pub use crate::internal::DEFAULT_ACCOUNT_ADDR;
+use casper_types::account::AccountHash;
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/lib.rs]: Let's merge this into the other `casper_types` imports.

File: execution_engine_testing/test_support/src/lib.rs [hunk 6]
```diff
@@ -65,35 +7,156 @@
...
 
 /// Default initial balance of a test account in motes.
 pub const DEFAULT_ACCOUNT_INITIAL_BALANCE: u64 = 100_000_000_000_000_000u64;
 
 /// Minimal amount for a transfer that creates new accounts.
 pub const MINIMUM_ACCOUNT_CREATION_BALANCE: u64 = 7_500_000_000_000_000u64;
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/lib.rs]: These two consts should be moved alongside the rest of the consts.

File: execution_engine_testing/test_support/src/lib.rs [hunk 7]
```diff
@@ -65,35 +7,156 @@
...
+/// Annual issuance: 2%
+/// Minimum round exponent: 14
+/// Ticks per year: 31536000000
+///
+/// (1+0.02)^((2^14)/31536000000)-1 is expressed as a fraction below.
+pub const DEFAULT_ROUND_SEIGNIORAGE_RATE: Ratio<u64> = Ratio::new_raw(6414, 623437335209);
+
+/// Default chain name.
+pub const DEFAULT_CHAIN_NAME: &str = "casper-execution-engine-testing";
+/// Default gensis timestamp in milliseconds.
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/lib.rs]: Typo in genesis.

File: execution_engine_testing/test_support/src/lib.rs [hunk 8]
```diff
@@ -65,35 +7,156 @@
...
+pub const DEFAULT_GENESIS_TIMESTAMP_MILLIS: u64 = 0;
+/// Default maximum number of associated keys.
+pub const DEFAULT_MAX_ASSOCIATED_KEYS: u32 = 100;
+/// Default block time.
+pub const DEFAULT_BLOCK_TIME: u64 = 0;
+/// Default gas price.
+pub const DEFAULT_GAS_PRICE: u64 = 1;
+/// Address of a mocked account.
+pub const MOCKED_ACCOUNT_ADDRESS: AccountHash = AccountHash::new([48u8; 32]);
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/lib.rs]: I believe this const can be removed.

File: execution_engine_testing/test_support/src/lib.rs [hunk 9]
```diff
@@ -65,35 +7,156 @@
...
+/// Amount named argument.
+pub const ARG_AMOUNT: &str = "amount";
+/// TImestamp increment in milliseconds.
+pub const TIMESTAMP_MILLIS_INCREMENT: u64 = 30000; // 30 seconds
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/lib.rs]:
```suggestion
/// Timestamp increment in milliseconds.
pub const TIMESTAMP_MILLIS_INCREMENT: u64 = 30_000; // 30 seconds
```

File: execution_engine_testing/test_support/src/lib.rs [hunk 10]
```diff
@@ -65,35 +7,156 @@
...
+
+// NOTE: Those values could be constants but are kept as once_cell::sync::Lazy to avoid changes of
+// `*FOO` into `FOO` back and forth.
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/lib.rs]:
```suggestion
```

File: execution_engine_testing/test_support/src/upgrade_request_builder.rs [hunk 11]
```diff
@@ -6,6 +6,7 @@ use casper_execution_engine::core::engine_state::UpgradeConfig;
 use casper_hashing::Digest;
 use casper_types::{EraId, Key, ProtocolVersion, StoredValue};
 
+/// TODO: doc comment.
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/upgrade_request_builder.rs]: Need doc comments throughout the file.

File: execution_engine_testing/test_support/src/step_request_builder.rs [hunk 12]
```diff
@@ -5,6 +5,7 @@ use casper_execution_engine::core::engine_state::{
 use casper_hashing::Digest;
 use casper_types::{EraId, ProtocolVersion};
 
+/// TODO: doc comment.
```
Comment (Fraser999) [inline execution_engine_testing/test_support/src/step_request_builder.rs]: Need doc comments throughout the file.

File: execution_engine_testing/test_support/src/deploy_item_builder.rs (hunk 4 above)
Comment (nilz3ro) [reply execution_engine_testing/test_support/src/deploy_item_builder.rs]: 😅 oops

File: execution_engine_testing/test_support/src/step_request_builder.rs (hunk 12 above)
Comment (nilz3ro) [reply execution_engine_testing/test_support/src/step_request_builder.rs]: done

**Review by Fraser999 — APPROVED** [review]

**Comment by nilz3ro** [comment]
bors merge

**Pull request merged**